core/src/main/java/org/infinispan/CacheStream.java

```diff
@@ -180,12 +184,21 @@
     * wish to have a parallel variant you can use {@link java.util.stream.StreamSupport#stream(Spliterator, boolean)}
     * passing in the spliterator from the stream.  In either case remember you <b>must</b> close the stream after
     * you are done processing the iterator or spliterator..</p>
-    * @param action
+    * @param action consumer to be ran for each element in the stream
     */
    @Override
    void forEach(Consumer<? super R> action);
 
    /**
+    * Same as {@link CacheStream#forEach(Consumer)} except that the Consumer must also implement Serializable.
+    * <p>
+    * This method will be used automatically by lambdas, which prevents users from having to manually cast to
+    * a Serializable lambda.
```

Comment:
Yeah, I was trying to limit how much stuff I had to change on each copy/paste :)  I kinda figured the first paragraph would be fine then.